### PR TITLE
Feature/sa 178 actionable dashboards

### PR DESCRIPTION
**Pull request by progers-tech**

### CODE REVIEW

File: infrastructure/elastic_resources/dashboards.tf [hunk 1]
```diff
@@ -82,10 +82,119 @@ module "moj_dashboard_3" {
...
   object_title = "SSL Expiry"
   es_domain    = data.aws_ssm_parameter.es_domain.value
 }
 
+
+module "moj_dash_ai_open_ports" {
+  // two slashes are intentional: https://www.terraform.io/docs/modules/sources.html#modules-in-package-sub-directories
+  source = "github.com/ministryofjustice/securityanalytics-analyticsplatform//infrastructure/kibana_saved_object"
+
+  // It is sometimes useful for the developers of the project to use a local version of the task
```
Comment (blackmamo) [inline infrastructure/elastic_resources/dashboards.tf]: These need to be # not //

File: infrastructure/elastic_resources/searches.tf [hunk 2]
```diff
@@ -74,3 +74,28 @@ module "weak_ssl_proto_search" {
   es_domain    = data.aws_ssm_parameter.es_domain.value
 }
 
+
+module "open_ports_search" {
+  // two slashes are intentional: https://www.terraform.io/docs/modules/sources.html#modules-in-package-sub-directories
+  source = "github.com/ministryofjustice/securityanalytics-analyticsplatform//infrastructure/kibana_saved_object"
+
+  // It is sometimes useful for the developers of the project to use a local version of the task
+  // execution project. This enables them to develop the task execution project and the nmap scanner
```
Comment (blackmamo) [inline infrastructure/elastic_resources/searches.tf]: # not //

File: infrastructure/elastic_resources/visualisations_certs.tf [hunk 3]
```diff
@@ -48,3 +48,54 @@ module "ssl_expiry_table" {
...
 }
 
+
+module "ai_expiry_dates_filter" {
+  // two slashes are intentional: https://www.terraform.io/docs/modules/sources.html#modules-in-package-sub-directories
+  source = "github.com/ministryofjustice/securityanalytics-analyticsplatform//infrastructure/kibana_saved_object"
+
+  // It is sometimes useful for the developers of the project to use a local version of the task
+  // execution project. This enables them to develop the task execution project and the nmap scanner
+  // (or other future tasks), at the same time, without requiring the task execution changes to be
```
Comment (blackmamo) [inline infrastructure/elastic_resources/visualisations_certs.tf]: # not //

File: tests/integration_tests/nmap_integration_test.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 @pytest.mark.integration
 @pytest.mark.asyncio
 async def test_integration():
-    timeout = 240
+    timeout = 900
```
Comment (blackmamo) [inline tests/integration_tests/nmap_integration_test.py:14]: Comment saying yes an all port scan does take that long

File: tests/results_parser/results_parser_test.py [hunk 5]
```diff
@@ -209,6 +211,7 @@ def test_parses_hosts_and_ports():
                     ],
                     "ports": [
                         {
+                            "port_str": "22",
```
Comment (blackmamo) [inline tests/results_parser/results_parser_test.py:214]: I don't get this, why do we need port_id and port_str?

**Review by blackmamo — CHANGES_REQUESTED** [review]
# not //, plus some other observations/queustions also the branch needs rebasing. It is out of date

File: tests/results_parser/results_parser_test.py (hunk 5 above)
Comment (philr-dev) [reply tests/results_parser/results_parser_test.py]: port_str forces it to be a string so we don't have commas in the results output for open ports.
port_id is left there for anyone that's already using it (e.g. secondary scanners), as well as if at some point someone wants to do a range search by port

Comment (progers-tech) [reply tests/results_parser/results_parser_test.py:214]: port_str forces it to be a string so we don't have commas in the results output for open ports.
port_id is left there for anyone that's already using it (e.g. secondary scanners), as well as if at some point someone wants to do a range search by port

File: tests/integration_tests/nmap_integration_test.py (hunk 4 above)
Comment (progers-tech) [reply tests/integration_tests/nmap_integration_test.py:14]: done this :)

File: infrastructure/elastic_resources/dashboards.tf (hunk 1 above)
Comment (progers-tech) [reply infrastructure/elastic_resources/dashboards.tf]: all changed across the .tf files

File: results_parser/results_parser.py [hunk 6]
```diff
@@ -81,6 +81,7 @@ def _process_ports(self, ports, host, results_context):
                 port_id, protocol = (port['portid'], port['protocol'])
                 print(f"Looking at port: {(port_id, protocol)}")
                 port_key = {
+                    "port_str": str(port_id),
```
Comment (blackmamo) [inline results_parser/results_parser.py:87]: I still don't understand why we need both of these?

File: tests/results_parser/results_parser_test.py (hunk 5 above)
Comment (blackmamo) [reply tests/results_parser/results_parser_test.py:214]: Where were the commas coming from? Can we have a conversation about this one?

**Review by blackmamo — CHANGES_REQUESTED** [review]
I just want to ensure this port_str thing is a good idea first

**Review by blackmamo — APPROVED** [review]

**Pull request merged**